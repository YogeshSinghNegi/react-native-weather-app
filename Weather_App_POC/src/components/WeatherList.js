import * as AppColors from '../utils/AppColors';

import { SectionList, StyleSheet, Text } from 'react-native';

import ListItem from './ListItem';
import ListItemSeparator from '../components/ListItemSeparator';
import Moment from 'moment';
import React from 'react';

const WeatherList = (props) => {
    return (
        <SectionList
            sections = { props.data }
            keyExtractor = { (item, index) => item.dt_txt }
            renderItem = {({ item }) => (
              <ListItem item={item} />
            )}
            ItemSeparatorComponent = { ListItemSeparator }
            renderSectionHeader = {({ section }) => (
              <Text style={Styles.header}> 
                {Moment(section.key, 'YYYY-MM-DD').format('DD MMM, ddd')} 
              </Text>
            )}
        />
    );
};

const Styles = StyleSheet.create({
    header: {
        fontSize: 20,
        padding: 15,
        backgroundColor: AppColors.darkColor,
        color: AppColors.whiteColor,
      }
});

export default WeatherList;