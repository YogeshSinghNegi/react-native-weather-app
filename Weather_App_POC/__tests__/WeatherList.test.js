import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import WeatherList from '../src/components/WeatherList';

const weatherArr = [
   {
    "key":"2020-07-27",
    "data":{"dt":1589220000,"main":{"temp":305.15,"feels_like":301.8,"temp_min":305.15,"temp_max":306.2,"pressure":1007,"sea_level":1007,"grnd_level":983,"humidity":37,"temp_kf":-1.05},"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"clouds":{"all":55},"wind":{"speed":7.34,"deg":314},"rain":{"3h":0.32},"sys":{"pod":"n"},"dt_txt":"2020-07-27 18:00:00"}
   },{
    "key":"2020-07-27",
    "data":{"dt":1589220000,"main":{"temp":305.15,"feels_like":301.8,"temp_min":305.15,"temp_max":306.2,"pressure":1007,"sea_level":1007,"grnd_level":983,"humidity":37,"temp_kf":-1.05},"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"clouds":{"all":55},"wind":{"speed":7.34,"deg":314},"rain":{"3h":0.32},"sys":{"pod":"n"},"dt_txt":"2020-07-27 18:00:00"}
   }
];

configure({adapter: new Adapter()});

describe('Testing <WeatherList/> component', () => {
  it('renders as expected', () => {
    const dataList = shallow(
      <WeatherList data = {weatherArr} />
    );
    expect(dataList).toMatchSnapshot();
    dataList.setProps({ weatherData: [weatherArr] });
    expect(dataList).toMatchSnapshot();
    const flatList = dataList.find('SectionList');
    const item = flatList.renderProp('renderItem')({ item: weatherArr });
    const item1 = flatList.renderProp('renderSectionHeader')
    const item3 = flatList.renderProp('keyExtractor')({ item: weatherArr });
    
    expect(item3).toMatchSnapshot();
    expect(item).toMatchSnapshot();
    expect(item1).toMatchSnapshot();
  });
});