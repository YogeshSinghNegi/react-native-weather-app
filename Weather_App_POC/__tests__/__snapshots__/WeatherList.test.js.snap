// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <WeatherList/> component renders as expected 1`] = `ShallowWrapper {}`;

exports[`Testing <WeatherList/> component renders as expected 2`] = `ShallowWrapper {}`;

exports[`Testing <WeatherList/> component renders as expected 3`] = `ShallowWrapper {}`;

exports[`Testing <WeatherList/> component renders as expected 4`] = `ShallowWrapper {}`;

exports[`Testing <WeatherList/> component renders as expected 5`] = `[Function]`;
